docroot/profiles/custom/cgov_site/modules/custom/cgov_core/cgov_core.module

```diff
@@ -848,3 +852,46 @@ function cgov_core_taxonomy_term_delete(EntityInterface $entity) {
     }
   }
 }
+
+/**
+ * Implements hook_ENTITY_TYPE_insert().
+ */
+function cgov_core_redirect_insert($entity) {
+  $moduleHandler = \Drupal::service('module_handler');
+  if ($moduleHandler->moduleExists('purge')) {
+    // The value for cgdp.cache_url_host is computed in cgov_core.settings.php.
+    $host = Settings::get('cgdp.cache_url_host');
+    if (empty($host)) {
+      return;
+    }
+    $purgeInvalidationFactory = \Drupal::service('purge.invalidation.factory');
+    $purgeProcessors = \Drupal::service('purge.processors');
+    $purgePurgers = \Drupal::service('purge.purgers');
+    $processor = $purgeProcessors->get('drush_purge_invalidate');
+    if (empty($processor)){
+      return;
+    }
+    $language = [$entity->language->value];
+    $sourcePath = '/' . $entity->redirect_source->path;
+    $langcodes = \Drupal::languageManager()->getLanguages();
+    // If und set means redirect is created for all languages
+    // in that case we need to purge url for all languages
+    // installed in system.
+    if ($entity->language->value === 'und'){
+      $language = array_keys($langcodes);
+    }
+    $host = rtrim($host, '/');
+    $invalidations = [];
+    foreach ($language as $value) {
```

Comment:
And again, a comment explaining what this block of code is doing. Not a line-by-line breakdown, just the purpose of this loop.